### DISCUSSION TITLE
hub's  "source" jetstream get duplicate data

### DISCUSSION THREAD

**Discussion by changs1986** [nats-io/nats-server#5669] [Q&A]
i have a question about leafnodes and hub communication. please help me understand how to solve.

i have leafnode1 domain leaf1 and leafnode2 domain leaf2. and one hub server. They run on three different servers.  

leaf1 generates data like:
`topic: events.a.d payload: {"t": 123456}`

leaf2 generates data like:
`topic: events.c.d payload: {"t": 123456}`

i want save leaf1's own data on leaf1's jetstream j1 . leaf2's own data on leaf2's jetstream j2. j1 and j2 jetstream subjects using events.*.d to capture the data. and then hub uses jetstream source to aggregate j1's & j2's data.  but I found the hub is sending j1's data to j2, sending j2's data to j1. so the hub got's the DUPLICATE data on his jetstream.  my question is how to avoid this problem? how j1 and j2 keep its own server data.

**Comment by amitamit281** [comment]
Hi @changs1986 

This is natural because your both leaf nodes has interest in "events.*.d". So to avoid that you can add deny sub "events.*.d" in hub  creds that is connfigured in your leaf node to connect hub

leafnodes: {
    remotes=[
        {
        url: ["nats-local:7422"]
        account: <transition account id>
        credentials: "a.creds"
       }
    ]
}
